### PR TITLE
Operations refactoring

### PR DESCRIPTION
**Pull request by slemrmartin**
In this repo this PR bumps providers-common, do some renaming and uses providers-common sources/topological APIs

---

**Depends on**:

- [x] https://github.com/RedHatInsights/topological_inventory-providers-common/pull/52
- [x] https://github.com/RedHatInsights/topological_inventory-providers-common/pull/53

---

[RHCLOUD-9343](https://issues.redhat.com/browse/RHCLOUD-9343)

### CODE REVIEW

**Comment by slemrmartin** [comment]
@syncrou ( cc @lindgrenj6 ) 
Is it worth to main ServicePlan#order operation? It needs complete refactoring but I doubt it's intended to be used in the future

**Comment by syncrou** [comment]
@slemrmartin correct. I don't see the point in making any changes to the ordering process. The long term plan is to abandon it, short term I'm not sure. Let's leave it where it is at for now until we know more.

File: lib/topological_inventory/openshift/operations/service_catalog_client.rb [hunk 1]
```diff
@@ -0,0 +1,92 @@
+require 'more_core_extensions/core_ext/hash'
+require "topological_inventory/openshift/logging"
+require "topological_inventory/openshift/connection"
+require "topological_inventory/providers/common/mixins/sources_api"
+require "topological_inventory/providers/common/mixins/topology_api"
+
+module TopologicalInventory
+  module Openshift
+    module Operations
+      class ServiceCatalogClient
```
Comment (slemrmartin) [inline lib/topological_inventory/openshift/operations/service_catalog_client.rb:10]: moved from `Operations::Code::ServiceCatalogClient`

File: lib/topological_inventory/openshift/operations/order/request.rb [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+          attr_accessor :operation, :params, :identity, :metrics
+
+          def initialize(params, identity, metrics = nil, operation_name = 'ServicePlan#order')
+            self.operation = operation_name
+            self.metrics = metrics
+            self.params = params
+            self.identity = identity
+          end
+
+          def run
```
Comment (slemrmartin) [inline lib/topological_inventory/openshift/operations/order/request.rb:21]: moved from `ServicePlan.order`

**Review by lindgrenj6 — APPROVED** [review]

**Pull request merged**